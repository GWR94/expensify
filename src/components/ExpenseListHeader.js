import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenseTotal from '../selectors/expense-total';
import selectExpenses from '../selectors/expenses';

export const ExpenseListHeader = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount === 1 ? "expense" : "expenses";
    const formattedTotal = numeral(expenseTotal / 100).format('$0,0.00');
    return (
        <div>
            { expenseCount > 0 ? (
                <h3>You have {expenseCount} {expenseWord}, totalling {formattedTotal}</h3>
            ) : (
                <h3>You have no current expenses</h3>
            )
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpenseTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseListHeader);