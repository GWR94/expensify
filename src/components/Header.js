import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="active" exact={true}>Home</NavLink></li>
            <li><NavLink to="/create" activeClassName="active">Add Expense</NavLink></li>
            <li><NavLink to="/help" activeClassName="active">Help</NavLink></li>u
        </ul>
    </header>
);

export default Header;