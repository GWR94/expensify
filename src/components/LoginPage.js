import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control</p>
      <button className="button__login button__login--google" onClick={startLoginGoogle}><i className="fa fa-google" aria-hidden="true"></i>&nbsp;&nbsp;Login With Google</button>
      <button className="button__login button__login--github"><i className="fa fa-github" aria-hidden="true"></i>&nbsp;&nbsp; Login With Github</button>
      <button className="button__login button__login--facebook"><i className="fa fa-facebook" aria-hidden="true"></i>&nbsp;&nbsp; Login With Facebook</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
