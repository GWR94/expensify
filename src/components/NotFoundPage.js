import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
	<div>
		<div className="page-header">
			<div className="content-container">
				<h1 className="page-header__title">404 Error!</h1>
			</div>
		</div>
		<div className="content-container">
			<img src="../../public/images/404rickmorty.png" />
		</div>
	</div>
);

export default NotFoundPage;
