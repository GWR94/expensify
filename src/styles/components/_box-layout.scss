.box-layout {
    background: url('/images/bg.jpg');
    width: 100vh;
    height: 100vh;
}