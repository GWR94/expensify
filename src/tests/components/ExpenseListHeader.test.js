import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListHeader } from '../../components/ExpenseListHeader';

test('should render the ExpenseListHeader component with the correct value with 1 expense', () => {
    const wrapper = shallow(<ExpenseListHeader expenseCount={1} expenseTotal={3499} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render the ExpenseListHeader component with the correct value with multiple expenses', () => {
    const wrapper = shallow(<ExpenseListHeader expenseCount={2} expenseTotal={10299} />);
    expect(wrapper).toMatchSnapshot();
});