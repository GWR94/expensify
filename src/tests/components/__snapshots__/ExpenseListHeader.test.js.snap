// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpenseListHeader component with the correct value with 1 expense 1`] = `
<div>
  <h3>
    You have 
    1
     
    expense
    , totalling 
    $34.99
  </h3>
</div>
`;

exports[`should render the ExpenseListHeader component with the correct value with multiple expenses 1`] = `
<div>
  <h3>
    You have 
    2
     
    expenses
    , totalling 
    $102.99
  </h3>
</div>
`;
